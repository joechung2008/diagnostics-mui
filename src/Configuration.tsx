import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Typography,
} from "@mui/material";

const Configuration: React.FC<ConfigurationProps> = ({ config }) => {
  const items = Object.entries(config).reduce<KeyValuePair<string>[]>(
    (previous, [key, value]) => [...previous, { key, value }],
    []
  );

  return (
    <div>
      <Typography variant="h3" fontWeight="semibold">
        Configuration
      </Typography>
      <Table aria-label="Configuration">
        <TableHead>
          <TableRow>
            <TableCell component="th">Key</TableCell>
            <TableCell component="th">Value</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {items.map((item) => (
            <TableRow key={item.key}>
              <TableCell>{item.key}</TableCell>
              <TableCell>{item.value}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
};

export default Configuration;
