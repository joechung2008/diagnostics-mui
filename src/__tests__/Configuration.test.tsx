import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Configuration from "../Configuration";

describe("Configuration", () => {
  it("matches snapshot", () => {
    const { container } = render(<Configuration config={{ key: "value" }} />);
    expect(container).toMatchSnapshot();
  });
});
