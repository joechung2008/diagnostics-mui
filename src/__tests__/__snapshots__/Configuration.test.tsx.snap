// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Configuration > matches snapshot 1`] = `
<div>
  <div>
    <h3
      class="MuiTypography-root MuiTypography-h3 css-1lw8d58-MuiTypography-root"
    >
      Configuration
    </h3>
    <table
      aria-label="Configuration"
      class="MuiTable-root css-1xwxv7r-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
            scope="col"
          >
            Key
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
            scope="col"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
          >
            key
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
          >
            value
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
